bioptim/dynamics/configure_problem.py

```diff
@@ -312,6 +316,68 @@ def configure_contact_function(ocp, nlp, dyn_func: Callable, **extra_params):
             legend=all_contact_names,
         )
 
+    @staticmethod
+    def configure_soft_contact_function(ocp, nlp):
+        """
+        Configure the soft contact sphere
+
+        Parameters
+        ----------
+        ocp: OptimalControlProgram
+            A reference to the ocp
+        nlp: NonLinearProgram
+            A reference to the phase
+        """
+        nlp.soft_contact_forces_func = []
+        for i_sc in range(nlp.model.nbSoftContacts()):
+            soft_contact = nlp.model.softContact(i_sc)
+            n = int(nlp.states.mx_reduced.shape[0] / 2)
+            softcontact_force_func = (
+                biorbd.SoftContactSphere(soft_contact)
+                .computeForceAtOrigin(nlp.model, nlp.states.mx_reduced[:n], nlp.states.mx_reduced[n:])
+                .to_mx()
+            )
+
+            nlp.soft_contact_forces_func.append(
+                Function(
+                    "soft_contact_forces_func",
+                    [nlp.states.mx_reduced, nlp.controls.mx_reduced, nlp.parameters.mx],
+                    [softcontact_force_func],
+                    ["x", "u", "p"],
+                    ["soft_contact_forces"],
+                ).expand()
+            )
+
+            all_soft_contact_names = []
+            l = ["Mx", "My", "Mz", "Fx", "Fy", "Fz"]
+            all_soft_contact_names.extend(
+                [
+                    f"{nlp.model.softContactName(i_sc).to_string()}_{name}"
+                    for name in l
+                    if nlp.model.softContactName(i_sc).to_string() not in all_soft_contact_names
+                ]
+            )
+
+            if "soft_contact_forces" in nlp.plot_mapping:
+                phase_mappings = nlp.plot_mapping["soft_contact_forces"]
+            else:
+
+                soft_contact_names_in_phase = [
+                    f"{nlp.model.softContactName(i_sc).to_string()}_{name}"
+                    for name in l
+                    if nlp.model.softContactName(i_sc).to_string() not in all_soft_contact_names
+                ]
+                phase_mappings = Mapping(
+                    [i for i, c in enumerate(all_soft_contact_names) if c in soft_contact_names_in_phase]
+                )
+
+            nlp.plot["soft_contact_forces"] = CustomPlot(
+                lambda t, x, u, p: nlp.soft_contact_forces_func[i_sc](x, u, p),
```

Comment:
I think this is a bug. You should not interface a list in a lambda function otherwise it instanciate it once and then reuse the same all the time leading to a single graph repeated n times

Reply:
I understand, i created only one plot for last soft contact, because previous ones are replaced each time by the loop.